bin/nvram-compact

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/perl
+# Remove invalid and non-data entries from an NVRAM section
+# This only handles the common cases
+
+use strict;
+use warnings;
+use FindBin;
+use lib "$FindBin::Bin/../lib";
+use EFI;
+use Getopt::Long;
+use File::Basename;
+use File::Temp 'tempfile';
+use Data::Dumper;
+
+my $nvram_guid = "cef5b9a3-476d-497f-9fdc-e98143e0422c";
+my $verbose = 0;
+
+local $/;
+my $ffs = <>;
+
+my $file_guid = EFI::read_guid($ffs, 0x00);
+die "$file_guid != nvram guid $nvram_guid\n"
+	unless $file_guid eq $nvram_guid;
+
+# should check for FFS header 2, as well as confirm length
+my $offset = 0x18;
+my $ffs_hdr = substr($ffs, 0x00, $offset);
+my $ffs_len = length($ffs);
+my $data = '';
+
+
+
+while($offset < $ffs_len - 0x08)
+{
+	my $nvar = EFI::NVRAM::NVAR->parse($ffs, $offset)
+		or last;
+
+	$offset += $nvar->length();
+
+	# Decide if we keep it:
+	# No valid bit? do not keep
+	next unless $nvar->valid();
+
+	# if this is not an ASCII name variable
+	# we probably have already copied its data by
+	# following the linked list
+	my $name = $nvar->name()
```

Comment:
is it guaranteed that the other entries in the list besides the first one will not have a name? If not this could lead to two copies of the same variable